lib/base/configobject.cpp

```diff
@@ -468,6 +468,12 @@ void ConfigObject::DumpObjects(const String& filename, int attributeTypes)
 	Log(LogInformation, "ConfigObject")
 		<< "Dumping program state to file '" << filename << "'";
 
+	try {
+		Utility::Glob(filename + ".*", &Utility::Remove, GlobFile);
```

Comment:
This might have undesirable side-effects for example if someone copied `icinga2.state` to for example `icinga2.state.bak` for whatever reason. So maybe we should change the file name to `icinga2.state.tmp.XXXXXX` to allow more specific matching? Won't catch any already existing files then though.